File: projects/m1/003-making-change/solutions/valentinaBazzacchi/solution-making-change.js
/*
toonie= 200;
loonie= 100;
quarter= 25;
dime= 10;
nickel= 5;
penny= 1;
*/

let changeAmount; 
let toonie;
let loonie;
let quarter;
let dime;
let nickel;
let penny;
let remainder;


changeAmount= +prompt("How many cents do you have to give to the shopper?")*100;

toonie= Math.floor(changeAmount/200);
remainder= changeAmount % 200;

loonie= Math.floor(remainder/100);
remainder= remainder % 100;

quarter= Math.floor(remainder/25);
remainder= remainder % 25;

dime= Math.floor(remainder/10);
remainder= remainder % 10;

nickel= Math.floor(remainder/5);
remainder= remainder % 5;

penny= Math.floor(remainder/1);
remainder= remainder % 1;

alert( `You should give to the shopper these coins: 
${toonie+ (" toonies")}, ${loonie+ (" loonies")}, ${quarter+ (" quarters" )}, ${dime+ (" dimes" )}, ${nickel+ (" nickels" )}, ${penny+ (" pennies.")}` );

File: projects/m1/003-making-change/solutions/stefanoalfeo/solution-making-change.py
import math
read_penny = int(input("number of penny read\n"))

# definizione valore monete
penny = 1
nickel = 5
dime = 10
quarter = 25
loonie = 100
toonie = 200

print("total change:")
# inizio con toonie
if read_penny > toonie:
    coin = toonie
    change = read_penny / coin
    next_change = read_penny % coin
    print("toonies: " + str(math.trunc(change)))
else:
    next_change = read_penny

# passo a loonie
coin = loonie
if(next_change >= coin):
    change = next_change / coin
    next_change = next_change % coin
    print("loonies: " + str(math.trunc(change)))

# passo a quarter
coin = quarter
if(next_change >= coin):
    change = next_change / coin
    next_change = next_change % coin
    print("quarters: " + str(math.trunc(change)))

# passo a dime
coin = dime
if(next_change >= coin):
    change = next_change / coin
    next_change = next_change % coin
    print("dimes: " + str(math.trunc(change)))

# passo a nickel
coin = nickel
if(next_change >= coin):
    change = next_change / coin
    next_change = next_change % coin
    print("nickel: " + str(math.trunc(change)))

# passo a penny
coin = penny
if(next_change >= coin):
    change = next_change / coin
    next_change = next_change % coin
    print("penny: " + str(math.trunc(change)))